clang/lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -825,6 +838,86 @@ void MSVCToolChain::AddHIPIncludeArgs(const ArgList &DriverArgs,
   RocmInstallation.AddHIPIncludeArgs(DriverArgs, CC1Args);
 }
 
+#ifdef ENABLE_CLASSIC_FLANG
+void MSVCToolChain::AddFortranStdlibLibArgs(const ArgList &Args,
+                                    ArgStringList &CmdArgs) const {
+ bool staticFlangLibs = false;
+ bool useOpenMP = false;
+
+  if (Args.hasArg(options::OPT_staticFlangLibs)) {
+    for (auto *A: Args.filtered(options::OPT_staticFlangLibs)) {
+      A->claim();
+      staticFlangLibs = true;
+    }
+  }
+
+  Arg *A = Args.getLastArg(options::OPT_mp, options::OPT_nomp,
+                           options::OPT_fopenmp, options::OPT_fno_openmp);
+  if (A &&
+      (A->getOption().matches(options::OPT_mp) ||
+       A->getOption().matches(options::OPT_fopenmp))) {
+      useOpenMP = true;
+  }
+
+  if (needFortranMain(getDriver(), Args)) {
+    // flangmain is always static
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/subsystem:console");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:flangmain.lib");
+  }
+
+  if (staticFlangLibs) {
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:libflang.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:libflangrti.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:libpgmath.lib");
+  } else {
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:flang.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:flangrti.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:pgmath.lib");
+  }
+  if (useOpenMP) {
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/nodefaultlib:vcomp.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/nodefaultlib:vcompd.lib");
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back("/defaultlib:libomp.lib");
+  }
+  else {
+    if (staticFlangLibs) {
+      CmdArgs.push_back("-linker");
+      CmdArgs.push_back("/defaultlib:libompstub.lib");
+    } else {
+      CmdArgs.push_back("-linker");
+      CmdArgs.push_back("/defaultlib:ompstub.lib");
+    }
+  }
+
+  // Allways link Fortran executables with Pthreads
+  // CmdArgs.push_back("-lpthread");
+
+  // These options are added clang-cl in Clang.cpp for C/C++
+  // In clang-cl.exe -MD and -MT control these options, but in
+  // flang.exe like clang.exe these are different options for
+  // dependency tracking. Let's assume that if somebody needs
+  // static flang libs, they need static runtime libs as well.
```

Comment:
@kaadam, we have to make a decision on how to build static libraries. Should static libraries be built with static CRT (libcmt) or dynamic CRT (msvcrt)?
It seems like `libflang.lib` was built with dynamic CRT, but we are using `libcmt.lib` to link with static CRT which doesn't work.

Reply:
Yes, msvcrt conflicts with lbcmt. In general we built all runtime libs with /MD by default. As I see from the Windows documentation we should use /MT for static libs [1], I don't know whether it has any side effect if we don't use it. In this case we should tell to cmake to use /MT during static lib build via MSVC_RUNTIME_LIBRARY generator expression.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170#c-runtime-lib-files

Reply:
For now I think maybe better if we use 'msvcrt.lib' instead of libcmt. We could handle it in driver easier. So +1 to me for msvcrt at the moment.

Reply:
@isuruf HI, Could you update your PR to use msvcrt instead of libcmt in static? Thanks